backend-rust/src/graphql_api/block.rs

```diff
@@ -64,29 +64,35 @@ impl QueryBlocks {
                 FROM blocks
                 WHERE height > $1 AND height < $2
                 ORDER BY
-                    (CASE WHEN $4 THEN height END) DESC,
-                    (CASE WHEN NOT $4 THEN height END) ASC
+                    (CASE WHEN $4 THEN height END) ASC,
+                    (CASE WHEN NOT $4 THEN height END) DESC
                 LIMIT $3
-            ) ORDER BY height ASC",
+            ) ORDER BY height DESC",
             query.from,
             query.to,
             query.limit,
             query.desc
         )
-        .fetch(pool);
-
-        let mut connection = connection::Connection::new(true, true);
-        while let Some(block) = row_stream.try_next().await? {
-            connection.edges.push(connection::Edge::new(block.height.to_string(), block));
+        .fetch_all(pool)
+        .await?;
+        let has_prev_page = if let Some(first) = rows.first() {
+            sqlx::query_scalar!("SELECT true FROM blocks WHERE height > $1 LIMIT 1", first.height)
+                .fetch_optional(pool)
+                .await?
+                .flatten()
+                .unwrap_or_default()
+        } else {
+            false
+        };
+        let has_next_page = if let Some(last) = rows.last() {
+            last.height > 0
```

Comment:
Can we name 0 with something meaningful? 

For example: 
```
let genesis_height = 0;
last.height > genesis_height
```

Reply:
Added a comment